apps/fz_http/lib/fz_http/allow_rules.ex

```diff
@@ -4,23 +4,66 @@ defmodule FzHttp.AllowRules do
   """
 
   import Ecto.Query, warn: false
-  alias FzHttp.{AllowRules.AllowRule, Gateways.Gateway, Repo, Users.User}
+  alias FzHttp.{AllowRules.AllowRule, Gateways.Gateway, Repo, Users.User, Events, Telemetry}
 
   def create_allow_rule(attrs \\ %{}) do
-    %AllowRule{}
-    |> AllowRule.changeset(attrs)
-    |> Repo.insert()
+    case %AllowRule{}
+         |> AllowRule.changeset(attrs)
+         |> Repo.insert() do
+      {:ok, rule} ->
+        Events.add(rule)
+        Telemetry.add_rule()
+        {:ok, rule}
+
+      error ->
+        error
+    end
```

Comment:
We can make it more pretty if we add default `\\ %AllowRule{}` to the changeset first argument and then leverage with block:
```suggestion
    changeset = AllowRule.changeset(attrs) 
    
    with {:ok, allow_rule} <- Repo.insert(changeset) do
      Events.add(rule)
      Telemetry.add_rule()
      {:ok, allow_rule}
    end
```

Reply:
`AllowRule.changeset` still takes 2 arguments so I still need to keep the

```elixir
%AllowRule{}
|> AllowRule.changeset(attrs)
|> Repo.insert()
```

But I can leverage the `with` statement.